src/Mocking.jl

```diff
@@ -40,17 +40,17 @@ struct Patch{T}
 end
 
 macro patch(expr::Expr)
+    def = splitdef(expr, throw=false)
+
     # Expect a named function in long-form or short-form
-    if expr.head === :function || (expr.head === :(=) && expr.args[1].head == :call)
-        target_name = expr.args[1].args[1]
-        params = expr.args[1].args[2:end]
-        body = expr.args[2]
-    elseif expr.head == :(->)
-        throw(ArgumentError("expression needs to be a named function"))
-    else
+    if def === nothing
         throw(ArgumentError("expression is not a function definition"))
+    elseif def[:type] == :(->)
```

Comment:
can we not use the word `type` for something not releated to the typesytem?
Maybe  `kind`, or `form` ?

Also should we detect this error condition rather via

```suggestion
    elseif !haskey(def, :name)
```

Since that is what we are after?

Reply:
I think `:form` could be a good alternative. I agree `:type` is rather over used. Hmm, at first I didn't like the `haskey` idea but it's growing on me fast.

Reply:
I'm supportive of calling it `head` if it means `expr.head`

Reply:
> can we not use the word `type` for something not releated to the typesytem?
Maybe `kind`, or `form` ?

I think a different word is a good idea. I was going to go with `:form` but I like to use that to indicate long-form/short-form definition syntax. Since this is always the function definition head I think `:head` is quite reasonable.